fineract-doc/src/docs/en/chapters/features/buydown-fee.adoc

```diff
@@ -0,0 +1,637 @@
+
+- `buy.down.fee.not.enabled`: Buy Down Fee not enabled for loan product
+- `cannot.be.before.first.disbursement.date`: Invalid transaction date
+- `cannot.be.more.than.remaining.amount`: Adjustment exceeds balance
+- `loan.transaction.not.found`: Referenced transaction not found
+
+**Error Response Example**:
+[source,json]
+----
+{
+  "developerMessage": "Buy down fee is not enabled for this loan product",
+  "httpStatusCode": "400",
+  "defaultUserMessage": "Buy down fee is not enabled for this loan product",
+  "userMessageGlobalisationCode": "buy.down.fee.not.enabled",
+  "errors": [
+    {
+      "developerMessage": "Buy down fee is not enabled for this loan product",
+      "defaultUserMessage": "Buy down fee is not enabled for this loan product",
+      "userMessageGlobalisationCode": "buy.down.fee.not.enabled",
+      "parameterName": null
+    }
+  ]
+}
+----
+
+== Business Logic
+
+=== Amortization Process
+
+Buy Down Fee amortization follows the Equal Amortization strategy:
+
+1. **Initial Recording**: Buy down fee is recorded as a liability (Deferred Capitalized Income)
+2. **Daily Amortization**: Fee is amortized daily over the loan term
+3. **Income Recognition**: Amortized portions are recognized as income (Fee or Interest based on configuration)
+4. **COB Processing**: Close of Business processes handle automatic amortization
+5. **Balance Tracking**: System maintains current deferred fee balance at loan level
+6. **Customer Invisibility**: Amortization process is not visible to customers
+
+=== Special Scenarios
+
+==== Early Payoff Acceleration
+When a loan is paid off early, the system accelerates the remaining balance of the deferred income to ensure proper revenue recognition.
+
+==== Charge-off Handling
+When a loan balance is charged off, any remaining balance in deferred income is also charged off to maintain accounting accuracy.
+
+==== Deferred Fee Balance Management
+The system maintains a current deferred fee balance at the loan level, tracking:
+* Original fee amount
+* Amortized amount to date
+* Remaining balance to be amortized
+* Any adjustments or charge-offs
+
+=== Journal Entries
+
+==== Buy Down Fee Transaction
+```
+Dr. Buy Down Expense Account          [Fee Amount]
+    Cr. Deferred Capitalized Income       [Fee Amount]
+```
+
+==== Daily Amortization
+```
+Dr. Deferred Capitalized Income       [Daily Amortization]
+    Cr. Fee/Interest Income Account       [Daily Amortization]
+```
+
+==== Buy Down Fee Adjustment
+```
+Dr. Deferred Income Liability       [Adjustment Amount]
+    Cr. Buy Down Expense Account        [Adjustment Amount]
+```
+
+==== Buy Down Fee Amortization Adjustment
+```
+Dr. Deferred Income Liability       [Amortization Adjustment]
+    Cr. Income from Buy Down Account    [Amortization Adjustment]
+```
+
+==== Charge-off Handling
+```
+Dr. Losses Written Off Account      [Remaining Unamortized Amount]
+    Cr. Deferred Income Liability       [Remaining Unamortized Amount]
```

Comment:
Debits Deferred income, and credits ChargeOff Expense or Charge Off Fraud Expense if the loan was marked as fraud